src/confirm-modal/confirm-modal.tsx

```diff
@@ -0,0 +1,46 @@
+import { h } from 'preact'
+import cn from 'classnames'
+
+// Components
+import { IconButton } from '../icon-button/icon-button'
+
+// Types
+import type { IconButtonProps } from '../icon-button/icon-button'
+import type { ComponentChildren } from 'preact'
+
+// Style
+import classes from './confirm-modal.module.css'
+
+export type ConfirmModalProps = {
+	cancel?: IconButtonProps
+	confirm?: IconButtonProps
+	variant?: 'action' | 'danger'
+	children: ComponentChildren
+}
+
+const CONFIRM_VARIANTS: Record<
+	NonNullable<ConfirmModalProps['variant']>,
+	Extract<IconButtonProps['variant'], 'confirmAction' | 'confirmDanger'>
+> = {
```

Comment:
Does this need to be typed this way? IMHO I would just define the CONFIRM_VARIANTS first and then the `variant` type in the `ConfirmModalProps` can be `keyof CONFIRM_VARIANTS`, right?